StyleCop.Analyzers/StyleCop.Analyzers/Helpers/XmlCommentHelper.cs

```diff
@@ -268,22 +267,109 @@ internal static string GetText(XmlTextSyntax textElement, bool normalizeWhitespa
                 return null;
             }
 
-            StringBuilder stringBuilder = StringBuilderPool.Allocate();
+            string result = string.Empty;
+
+            StringBuilder stringBuilder = null;
 
             foreach (var item in textElement.TextTokens)
             {
-                stringBuilder.Append(item);
+                if (result.Length == 0)
+                {
+                    result = item.ToString();
+                }
+                else
+                {
+                    if (stringBuilder == null)
+                    {
+                        stringBuilder = StringBuilderPool.Allocate();
+                        stringBuilder.Append(result);
+                    }
+
+                    stringBuilder.Append(item.ToString());
+                }
+            }
+
+            if (stringBuilder != null)
+            {
+                result = StringBuilderPool.ReturnAndFree(stringBuilder);
             }
 
-            string result = StringBuilderPool.ReturnAndFree(stringBuilder);
             if (normalizeWhitespace)
             {
-                result = Regex.Replace(result, @"\s+", " ");
+                result = result.NormalizeWhiteSpace();
             }
 
             return result;
         }
 
+        internal static string NormalizeWhiteSpace(this string text)
+        {
+            if (text == null)
+            {
+                return null;
+            }
+
+            int length = text.Length;
+
+            bool lastSpace = false;
+
+            bool diff = false;
+
+            foreach (char ch in text)
+            {
+                if (char.IsWhiteSpace(ch))
+                {
+                    if (lastSpace)
+                    {
+                        length--;
+                    }
+                    else
+                    {
+                        if (ch != ' ')
+                        {
+                            diff = true;
+                        }
+
+                        lastSpace = true;
+                    }
+                }
+                else
+                {
+                    lastSpace = false;
+                }
+            }
+
+            if (diff || (length != text.Length))
+            {
+                char[] buffer = new char[length];
```

Comment:
as you probably know, if this code was targeting .NET Core (or conditionally compiled thus) you could eliminate this `char[]` allocation by using `string.Create( ...)`

Reply:
also from IndexOfAnyWhitespace which would be vectorized.
https://github.com/dotnet/runtime/issues/77959

Reply:
Yes, but this project is targeting multiple runtimes.